runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaPipelineResult.java

```diff
@@ -70,6 +75,11 @@ public State waitUntilFinish(@Nullable Duration duration) {
     }
 
     final StateInfo stateInfo = getStateInfo();
+
+    if (listener != null && (stateInfo.state == State.DONE || stateInfo.state == State.FAILED)) {
```

Comment:
I'm trying to understand when would we hit this while state is neither Done nor FAILED. What is being protected by having this check?

Reply:
Oh, the status can be RUNNING, since it can be a timed waifForFinish().